### PR TITLE
Add deploy workflow

### PR DESCRIPTION
**Pull request by NickGraham101**
### Context

We require a workflow to deploy the cluster.

### Changed proposed in this PR

- Extended documentation on cert creation
- Added local environment variable
- Add deploy cluster job
- Add validate-terraform job

### Guidance to review

Test run completed for `platform-test` environment [here](https://github.com/DFE-Digital/teacher-services-cloud/actions/runs/3939953306)

### CODE REVIEW

File: .github/actions/deploy-environment/action.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: Deploy Environment
+
+inputs:
+  azure_credentials:
+    description: The GitHub access token
```
Comment (saliceti) [inline .github/actions/deploy-environment/action.yml:5]: Wrong description?

File: .github/actions/deploy-environment/action.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    required: true
+  environment_name:
+    description: Name of the environment in Make file
+    required: true
+  root_folder:
```
Comment (saliceti) [inline .github/actions/deploy-environment/action.yml:10]: I understand this action is only for cluster, so we could hard code it in the action and remove the input

File: .github/actions/deploy-environment/action.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+runs:
+  using: composite
+
+  steps:
+  - run: |
+      TERRAFORM_VERSION=$(awk '/{/{f=/^terraform/;next}f' terraform/terraform.tf | grep -o [0-9\.]*)
+      . config/platform-test.sh
+
+      if [ -z "$RESOURCE_GROUP_NAME" ]; then
+        echo "::error ::Failed to extract app_resource_group_name from $TFVARS"
```
Comment (saliceti) [inline .github/actions/deploy-environment/action.yml:23]: from $TFVARS -> from the confirguration?

File: .github/actions/deploy-environment/action.yml (hunk 1 above)
Comment (NickGraham101) [reply .github/actions/deploy-environment/action.yml:5]: Corrected

File: .github/actions/deploy-environment/action.yml (hunk 3 above)
Comment (NickGraham101) [reply .github/actions/deploy-environment/action.yml:23]: Copy paste error 😳 fixed the error message

File: .github/actions/deploy-environment/action.yml (hunk 2 above)
Comment (NickGraham101) [reply .github/actions/deploy-environment/action.yml:10]: It was intended to be generalisable on the assumption we'd deploy everything using a workflow in the future

**Comment by NickGraham101** [comment]
[Create AKS Cluster CD Pipeline](https://trello.com/c/KXxbepDX/69-create-aks-cluster-cd-pipeline)

**Pull request merged**